src/js/msp/MSPHelper.js

```diff
@@ -785,6 +785,37 @@ MspHelper.prototype.process_data = function(dataHandler) {
                     };
                     MODE_RANGES.push(modeRange);
                 }
+
+                // MODE_RANGES_EXTRA isn't support before 1.42
+                if (semver.lt(CONFIG.apiVersion, "1.42.0")){
+                    MODE_RANGES_EXTRA = [];
+
+                    for (var i = 0; i < modeRangeCount; i++){
+                        var modeRange = MODE_RANGES[i];
+                        
+                        var modeRangeExtra = {
+                            id: modeRange.id,
+                            modeLogic: 0,
+                            linkedTo: 0
+                        }
+                        MODE_RANGES_EXTRA.push(modeRangeExtra);
+                    }
+                }
+                break;
+
+            case MSPCodes.MSP_MODE_RANGES_EXTRA:
+                MODE_RANGES_EXTRA = []; // empty the array as new data is coming in
+
+                var modeRangeCount = (data.byteLength - 1) / data.readU8();
```

Comment:
This will need changing after https://github.com/betaflight/betaflight/pull/7543#discussion_r254636886 is done.

Reply:
👍